ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterAggregator.java

```diff
@@ -53,33 +54,59 @@
    * for each partition.
    * @throws IOException
    */
-  Pair<StatsSnapshot, StatsSnapshot> doWork(Map<String, String> statsWrappersJSON, StatsReportType type) throws IOException {
+  Pair<StatsSnapshot, StatsSnapshot> doWork(Map<String, String> statsWrappersJSON, StatsReportType type)
+      throws IOException {
+    Map<String, StatsWrapper> statsWrappers = new HashMap<>();
+    for (Map.Entry<String, String> statsWrapperJSON : statsWrappersJSON.entrySet()) {
+      try {
+        if (statsWrapperJSON != null && statsWrapperJSON.getValue() != null) {
+          StatsWrapper snapshotWrapper = mapper.readValue(statsWrapperJSON.getValue(), StatsWrapper.class);
+          statsWrappers.put(statsWrapperJSON.getKey(), snapshotWrapper);
+        }
+      } catch (Exception e) {
+        logger.error("Exception occurred while processing stats from {}", statsWrapperJSON.getKey(), e);
+        exceptionOccurredInstances.computeIfAbsent(type, key -> new ArrayList<>()).add(statsWrapperJSON.getKey());
+      }
+    }
+    return doWorkOnStatsWrapperMap(statsWrappers, type, false);
+  }
+
+  Pair<StatsSnapshot, StatsSnapshot> doWorkOnStatsWrapperMap(Map<String, StatsWrapper> statsWrappers,
+      StatsReportType type) throws IOException {
+    return doWorkOnStatsWrapperMap(statsWrappers, type, true);
+  }
+
+  Pair<StatsSnapshot, StatsSnapshot> doWorkOnStatsWrapperMap(Map<String, StatsWrapper> statsWrappers,
+      StatsReportType type, boolean removeExceptionOnType) throws IOException {
     StatsSnapshot partitionSnapshot = new StatsSnapshot(0L, new HashMap<>());
     Map<String, Long> partitionTimestampMap = new HashMap<>();
     StatsSnapshot rawPartitionSnapshot = new StatsSnapshot(0L, new HashMap<>());
-    exceptionOccurredInstances.remove(type);
-    for (Map.Entry<String, String> statsWrapperJSON : statsWrappersJSON.entrySet()) {
-      if (statsWrapperJSON != null && statsWrapperJSON.getValue() != null) {
+    if (removeExceptionOnType) {
+      exceptionOccurredInstances.remove(type);
```

Comment:
What was the purpose of this piece of code resetting exceptionOccuredInstances for a given type again? Sorry, I'm not up to date on this.

Reply:
Trying to not change anywhere unnecessary to this PR.